libs/platform/table/components/table-p13-dialog/columns/columns.component.ts

```diff
@@ -329,11 +320,11 @@ export class P13ColumnsDialogComponent implements Resettable, OnInit, OnDestroy
                 .pipe(debounceTime(20))
                 .subscribe(([showAll, searchQuery]) => {
                     this._filteredColumns = this._selectableColumns
+                        .sort((a, b) => a.order - b.order)
```

Comment:
i think instead of sort it you can sort it inside _initiateColumns method, with that way you can reduce manipulation on array.